### PR TITLE
Dashboard: add a "Sorties Cinéma" widget backed by TMDB

### PR DESCRIPTION
**Pull request by vincentBesseau**
### Description

New cinema service (server/services/cinema) fetching upcoming theatrical releases from TMDB, resolving each movie's actual regional release date, overview and trailer (with an English fallback when the requested Gladys language has none), and caching results for 30 minutes.

New "cinema" dashboard box: a responsive poster grid, click-through detail panel (synopsis, trailer, TMDB link), and a 15/30/60-day window setting. Follows the openweather service pattern; localized in en/fr/de.

## Forum

Forum: https://community.gladysassistant.com/t/widjet-sortie-cinema/10757

### Checklist

- [x] Tests pass: `cd server && npm run coverage` (Codecov requires 100% coverage on changed lines) and Cypress (`npm run cypress:run`) if the UI changed
- [x] Linter and prettier pass on both front and server (`npm run eslint`, `npm run prettier`)
- [x] No undocumented breaking change

## Summary by CodeRabbit

* **New Features**
  * Added a TMDB-powered cinema integration for browsing upcoming movie releases.
  * Added a configurable dashboard widget with 15-, 30-, or 60-day viewing ranges.
  * Added regional release filtering using two-letter country codes, defaulting to France.
  * Added movie posters, localized release dates, overviews, and trailer links.
  * Added setup screens for configuring the TMDB API key.
  * Added English, French, and German translations.
  * Added responsive movie details and loading placeholders.

* **Bug Fixes**
  * Prevented outdated results from replacing newer cinema data during refreshes.

### CODE REVIEW

File: server/services/cinema/index.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+   * @returns {Promise<Array>} Resolve with the list of upcoming movies, soonest release first.
+   * @example
+   * gladys.services.cinema.movies.getUpcoming({ language: 'fr-FR', region: 'FR', daysAhead: 30 });
+   */
+  async function getUpcoming(options = {}) {
+    if (!tmdbApiKey) {
+      throw new ServiceNotConfiguredError('TMDB API Key not found');
+    }
+    const language = options.language || 'fr-FR';
+    const region = options.region || 'FR';
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/services/cinema/index.js]: Fixed in 7407c863: added a real `cinema_region` field on the box (editor input, wired through the frontend query params, controller and service), so it's no longer hardcoded to FR — every non-French install can now set their own theatrical calendar. Also switched `formatDate` from `toISOString()`/UTC to the server's local time, which narrows the day-boundary mismatch for self-hosted instances (a full house-timezone fix would need this service to accept a house, which it doesn't today — flagging as a known remaining limitation rather than solving it here).

File: server/services/cinema/index.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        `region=${region}`,
+        'with_release_type=2|3',
+        `release_date.gte=${todayFormatted}`,
+        `release_date.lte=${lastDayFormatted}`,
+      ].join('&');
+      const url = `https://api.themoviedb.org/3/discover/movie?${params}`;
+      try {
+        const { data } = await axios.get(url);
+        const candidates = formatUpcomingMovies(data);
+        const enrichedMovies = await Promise.all(candidates.map((movie) => enrichMovie(movie, language, region)));
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/services/cinema/index.js]: Fixed in 7407c863: added `mapWithConcurrency` (server/services/cinema/lib/mapWithConcurrency.js) and bounded the detail-lookup fan-out to 5 in-flight requests instead of firing all 20-40 via a single Promise.all.

File: server/services/tmdb/controllers/tmdb.controller.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+module.exports = function CinemaController(getUpcoming) {
+  /**
+   * @api {get} /api/v1/service/cinema/movies/upcoming Get upcoming movies
+   * @apiName getUpcomingMovies
+   * @apiGroup Cinema
+   */
+  async function getUpcomingController(req, res) {
+    const movies = await getUpcoming({
+      language: TMDB_LANGUAGE_BY_GLADYS_LANGUAGE[req.user.language],
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/services/tmdb/controllers/tmdb.controller.js]: Fixed in 7407c863: the fallback now resolves to 'en-US' (via the shared FALLBACK_OVERVIEW_LANGUAGE constant) instead of 'fr-FR', both here and in getUpcoming's own default. Region is now also a real setting sent by the widget — see the reply on the region comment.

File: front/src/config/integrations/index.js [hunk 4]
```diff
@@ -42,5 +43,6 @@ pushAllWithType(devices, 'device');
 pushAllWithType(communications, 'communication');
 pushAllWithType(calendars, 'calendar');
 pushAllWithType(weathers, 'weather');
+pushAllWithType(movies, 'movies');
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply front/src/config/integrations/index.js:46]: You're right, thanks for the pointer to §6.1/§6.4. Fixed in 7407c863: dropped the invented "movies" type, deleted movies.json, and moved the cinema entry into weathers.json under the existing "weather" type (route is now /dashboard/integration/weather/cinema). That also fixes the non-admin visibility gap you flagged below — weather is already in HIDDEN_TYPES_FOR_NON_ADMIN_USERS, so the shared TMDB key is now admin-only like OpenWeather's, with no extra change needed.

File: front/src/config/integrations/movies.json [hunk 5]
```diff
@@ -0,0 +1,7 @@
+[
+  {
+    "key": "cinema",
+    "img": "/assets/integrations/cover/cinema.jpg",
+    "categories": ["multimedia"]
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply front/src/config/integrations/movies.json:6]: Fixed in 7407c863 (folded into the weathers.json entry, see the reply above) — added `"cloud": true`.

File: front/src/components/boxs/premieres/PremieresBox.jsx [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+  componentDidMount() {
+    this.refreshData();
+    this.interval = setInterval(this.refreshData, REFRESH_INTERVAL_MS);
+  }
+
+  componentDidUpdate(previousProps) {
+    if (previousProps.box.days_ahead !== this.props.box.days_ahead) {
+      this.refreshData();
+    }
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply front/src/components/boxs/premieres/PremieresBox.jsx:263]: Fixed in 7407c863: CinemaBox now connects `user` and calls `dayjs(...).locale(userLanguage)` before every `.format()` call (both the grid and the detail modal), with a defensive 'en' fallback since `dayjs().locale()` with no truthy argument is a getter, not a setter — passing `undefined` straight through during the brief window before `user` loads would have crashed .format().

File: front/src/components/boxs/premieres/PremieresBox.jsx [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+  );
+};
+
+class CinemaBoxComponent extends Component {
+  refreshData = async () => {
+    try {
+      await this.setState(prevState => ({ error: false, loading: !prevState.movies }));
+      const daysAheadParams = this.props.box.days_ahead ? { daysAhead: this.props.box.days_ahead } : undefined;
+      const movies = await this.props.httpClient.get('/api/v1/service/cinema/movies/upcoming', daysAheadParams);
+      this.setState({ movies, loading: false, error: false });
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply front/src/components/boxs/premieres/PremieresBox.jsx:233]: Fixed in 8af65eeb: added a refreshSequence counter, exactly as suggested — an older in-flight request now no-ops on settling if a newer one has started.

File: front/src/components/boxs/cinema/style.css [hunk 8]
```diff
@@ -0,0 +1,70 @@
+@keyframes skeletonPulse {
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply front/src/components/boxs/cinema/style.css]: Fixed in 8af65eeb: renamed to skeleton-pulse (kebab-case) and updated the three references.

File: front/src/routes/integration/all/tmdb/Tmdb.jsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+import IntegrationSubPageLayout from '../../../../components/integration/IntegrationSubPageLayout';
+
+const CinemaPage = ({ children, ...props }) => (
+  <IntegrationSubPageLayout title={<Text id="integration.cinema.title" />}>
+    <div class="card">
+      <div class="card-body">
+        <div
+          class={cx('dimmer', {
+            active: props.loading
+          })}
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply front/src/routes/integration/all/tmdb/Tmdb.jsx:28]: Fixed in 8af65eeb: the page now renders a danger alert when either the key load or the key save/start request fails (props.error, derived from cinemaGetApiKeyStatus/cinemaSaveApiKeyStatus === RequestStatus.Error), while loading still only gates the dimmer.

File: server/services/tmdb/index.js [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+   * @example
+   * const value = await getOrCompute('upcoming:fr-FR:FR', () => doSomething());
+   */
+  async function getOrCompute(cacheKey, computeValue) {
+    const cached = cache.get(cacheKey);
+    if (cached && cached.expiresAt > Date.now()) {
+      return cached.value;
+    }
+    const value = await computeValue();
+    cache.set(cacheKey, { value, expiresAt: Date.now() + CACHE_DURATION_IN_MS });
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/services/tmdb/index.js:108]: Partially addressed in 8af65eeb: region is now validated against ^[A-Z]{2}$ before it reaches the cache key or the TMDB URL, which bounds the practical key space to daysAhead(3) x region(<=676) x language(3, always mapped server-side) — no longer arbitrary/unbounded. I did not add proactive expiry sweeping or a hard entry-count eviction on top of that: with the key space now bounded and each entry a small JSON array (a handful of KB), worst case is on the order of a few MB for the process lifetime of a self-hosted single-family instance, which doesn't seem worth the added complexity of an LRU/TTL sweep here. Happy to add it if you feel otherwise.

File: server/services/cinema/index.js [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+   * @param {number} movieId - The TMDB movie id.
+   * @param {string} language - The TMDB locale to request (ex. 'fr-FR').
+   * @returns {Promise<object|null>} The raw TMDB movie details response, or null.
+   * @example
+   * const details = await getMovieDetails(1234, 'fr-FR');
+   */
+  async function getMovieDetails(movieId, language) {
+    try {
+      const url = `https://api.themoviedb.org/3/movie/${movieId}?api_key=${tmdbApiKey}&language=${language}&append_to_response=release_dates,videos`;
+      const { data } = await axios.get(url);
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/services/cinema/index.js]: Fixed in 8af65eeb: added a 10s timeout to both TMDB axios.get calls (discover and details), and a test asserting every request carries a positive numeric timeout.

File: server/test/services/tmdb/lib/formatUpcomingMovies.test.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+    expect(movies).to.deep.equal(expectedResult);
+  });
+  it('should return an empty array when there are no results', () => {
+    const movies = formatUpcomingMovies({ results: [] });
+    expect(movies).to.deep.equal([]);
+  });
+  it('should return an empty array when results is missing', () => {
+    const movies = formatUpcomingMovies({});
+    expect(movies).to.deep.equal([]);
+  });
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/test/services/tmdb/lib/formatUpcomingMovies.test.js:18]: Fixed in 8af65eeb: added formatUpcomingMovies(null) and the matching-region-but-no-release_dates-field case to the two test files.

File: front/src/routes/integration/all/cinema/actions.js [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+    store.setState({
+      tmdbApiKey: e.target.value
+    });
+  },
+  async getApiKey(state) {
+    store.setState({
+      cinemaGetApiKeyStatus: RequestStatus.Getting
+    });
+    try {
+      const variable = await state.httpClient.get('/api/v1/service/cinema/variable/TMDB_API_KEY');
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply front/src/routes/integration/all/cinema/actions.js]: I don't think this is specific to this PR, so I'm leaving it unfixed here rather than deviating unilaterally in one integration: this actions.js follows the exact same state.httpClient.get/post pattern every other integration's key-saving screen uses (OpenWeather, and every other cloud API key stored via the generic /api/v1/service/:name/variable/:key route), none of which enforce HTTPS or block redirects at the call site. Gladys already supports plain-HTTP self-hosted deployments as a documented mode, and enforcing HTTPS in one widget's client code wouldn'\''t change what protocol the admin actually configured LOCAL_API_URL with — it would just make this one integration behave inconsistently with the rest of the app. If this is worth doing, it reads like a HttpClient-level change applying to every integration at once, not something to special-case here. Happy to be pointed at prior discussion on this if it exists.

File: front/src/components/boxs/cinema/EditCinemaBox.jsx [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+      days_ahead: Number(e.target.value)
+    });
+  };
+  const updateRegion = e => {
+    const region = e.target.value
+      .toUpperCase()
+      .replace(/[^A-Z]/g, '')
+      .slice(0, 2);
+    props.updateBoxConfig(props.x, props.y, {
+      cinema_region: region
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply front/src/components/boxs/cinema/EditCinemaBox.jsx]: Good catch, real bug — confirmed by testing it live (clearing the field 422'd the whole dashboard save). Fixed in fdc22b85: EditCinemaBox now tracks the raw input in local component state, and only persists to box config when the value is '' or a complete 2-letter code — a mid-typed single letter is displayed but never written to cinema_region.

File: server/models/dashboard.js [hunk 15]
```diff
@@ -43,6 +43,15 @@ const boxSchema = Joi.object().keys({
...
   scene_status_features: Joi.object().pattern(Joi.string(), Joi.string()),
+  // cinema box: how many days ahead to look for upcoming movie releases, and
+  // in which country's theatrical calendar (ISO 3166-1 alpha-2, ex. 'FR').
+  // Uppercase only: Joi's `.uppercase()` transform isn't applied back to the
+  // stored value here (the isEven() validator below only checks for errors,
+  // it doesn't persist the converted value), so a lowercase input would be
+  // accepted but stored mismatched — the pattern rejects it outright instead,
+  // and the editor UI always sends an uppercased value.
+  days_ahead: Joi.number().valid(15, 30, 60),
+  cinema_region: Joi.string().pattern(/^[A-Z]{2}$/),
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/models/dashboard.js]: Fixed in fdc22b85: cinema_region now allows '', matching weather's provider convention exactly as suggested. Added dashboard.update coverage for both the region-carrying and empty-region paths, plus a rejection test for a malformed value ('USA').

File: front/src/components/boxs/cinema/EditCinemaBox.jsx (hunk 14 above)
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply front/src/components/boxs/cinema/EditCinemaBox.jsx]: Same fix as the first pass on this finding — see fdc22b85 and the reply above.

File: server/models/dashboard.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/models/dashboard.js]: Same fix as the first pass on this finding — see fdc22b85 and the reply above.

File: server/services/cinema/index.js [hunk 16]
```diff
@@ -0,0 +1,218 @@
...
+   * @returns {Promise<Array>} Resolve with the list of upcoming movies, soonest release first.
+   * @example
+   * gladys.services.cinema.movies.getUpcoming({ language: 'fr-FR', region: 'FR', daysAhead: 30 });
+   */
+  async function getUpcoming(options = {}) {
+    if (!tmdbApiKey) {
+      throw new ServiceNotConfiguredError('TMDB API Key not found');
+    }
+    const language = options.language || FALLBACK_OVERVIEW_LANGUAGE;
+    const region = REGION_PATTERN.test(options.region) ? options.region : DEFAULT_REGION;
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/services/cinema/index.js]: Fixed in b8ca45d0: replaced the bare regex with isAssignedRegion (server/services/cinema/lib/isAssignedRegion.js), which uses Node's built-in Intl.DisplayNames to check the code resolves to a real region name rather than falling back to the code itself or 'Unknown Region' — no embedded ISO list needed. 'ZZ' and 'XX' both now fall back to DEFAULT_REGION; added a dedicated ZZ regression test plus a unit-test file for the new helper.

File: front/src/routes/integration/all/cinema/index.js [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+class CinemaIntegration extends Component {
+  componentWillMount() {
+    this.props.getApiKey();
+  }
+
+  render(props, {}) {
+    const loading =
+      props.cinemaSaveApiKeyStatus === RequestStatus.Getting || props.cinemaGetApiKeyStatus === RequestStatus.Getting;
+    const error =
+      props.cinemaSaveApiKeyStatus === RequestStatus.Error || props.cinemaGetApiKeyStatus === RequestStatus.Error;
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply front/src/routes/integration/all/cinema/index.js]: Fixed in b8ca45d0: saveApiKey now resets cinemaGetApiKeyStatus to Success when it starts, so a stale load error from getApiKey() no longer keeps the error alert visible after a later successful save. I did not add a Cypress regression test for this specific sequence — this repo's front has no unit-test runner (per AGENTS.md, front is covered by eslint/build/Cypress only) and a full E2E spec for this one status-clearing edge case felt disproportionate to add in this PR; happy to add one if you'd rather have it locked in that way.

File: server/services/cinema/lib/isAssignedRegion.js [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+ */
+function isAssignedRegion(code) {
+  if (!REGION_PATTERN.test(code)) {
+    return false;
+  }
+  // Unassigned/reserved codes are handed back unresolved by Intl.DisplayNames:
+  // either as the literal input ('XX') or as its dedicated "Unknown Region"
+  // label ('ZZ', the ISO-reserved "unknown or invalid territory" code).
+  const name = regionDisplayNames.of(code);
+  return name !== code && name !== 'Unknown Region';
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/services/cinema/lib/isAssignedRegion.js]: Good catch — verified: EU/UN/EZ do resolve to real display names via Intl.DisplayNames, so my previous fix was insufficient. Fixed in b283b684 by dropping the Intl-based heuristic entirely (also non-deterministic across Node/ICU versions, which isn't a great property for something feeding a cache key + a third-party URL) in favor of a static set sourced directly from TMDB's own /configuration/countries endpoint. That's the actual source of truth for whether region=X will match anything, and it also correctly keeps TMDB's legacy transitional codes (SU, YU, ZR...) that older catalog entries still carry, which a generic ISO 3166-1 list wouldn't have. Added regression tests for EU/UN/EZ rejection and SU/YU acceptance.

File: server/lib/premieres/premieres.getUpcoming.js [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+  }
+  // if every provider is merely not configured, the frontend must show its
+  // "configure a movie integration" call to action; a real failure (a
+  // provider exists but is broken) is more actionable and wins
+  let firstRealError = null;
+  const movies = await Promise.reduce(
+    candidates,
+    async (result, serviceName) => {
+      if (result !== null) {
+        return result;
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/lib/premieres/premieres.getUpcoming.js:51]: Fixed in 624171c0: added a test where an unpinned first-alphabetical provider succeeds while a second candidate exists, asserting the second is never called — this exercises the `if (result !== null) return result` short-circuit. Codecov patch is back to 100% (confirmed locally too).

File: server/lib/premieres/premieres.getUpcoming.js [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+          firstRealError = e;
+        }
+        return null;
+      }
+    },
+    null,
+  );
+  if (movies === null) {
+    if (firstRealError !== null) {
+      throw firstRealError;
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/lib/premieres/premieres.getUpcoming.js:77]: Good catch, real gap. Fixed in 4e07955e: cinema.getUpcoming now maps a firstRealError that is an ExternalIntegrationUnavailableError to Error400(REQUEST_TO_THIRD_PARTY_FAILED), same translation as weather.get. Added a test asserting an ext-* candidate's ExternalIntegrationUnavailableError surfaces as REQUEST_TO_THIRD_PARTY_FAILED.

**Comment by vincentBesseau** [comment]
documentation : https://github.com/GladysAssistant/v4-website/pull/409

File: server/lib/external-integration/externalIntegration.normalizeMovies.js [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+  if (id === null || title === null || releaseDate === null) {
+    return null;
+  }
+  const movie = { id, title, releaseDate };
+  OPTIONAL_STRING_FIELDS.forEach((field) => {
+    const value = toBoundedString(rawMovie[field]);
+    if (value !== null) {
+      movie[field] = value;
+    }
+  });
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/lib/external-integration/externalIntegration.normalizeMovies.js]: Fixed in 90c6524d: normalizeMovies now allowlists http(s) only for posterUrl/trailerUrl/sourceUrl (anything else, e.g. javascript:/data:, is dropped like any other invalid field), and rejects an oversized or non-finite id instead of truncating it — truncating an id risks silently rewriting it into a different, possibly colliding one, unlike a title.

File: docs/specs/external-integrations.md [hunk 22]
```diff
@@ -408,6 +408,24 @@ The core widget renders the base fields as today and the extensions **when prese
...
+
+**5. Provider-agnostic widget**: `PremieresBox.jsx` used to hardcode a `themoviedb.org` link on every entry — harmless while `tmdb` was the only provider, but wrong for any other one. It now renders `movie.sourceUrl` when present instead; `tmdb` was updated to populate that same generic field with its own `themoviedb.org` URL, so its own rendering is unchanged.
+
+**What this type deliberately does not have yet**: no provider-pin route equivalent to `GET /api/v1/weather/provider` (the existing `GET /api/v1/premieres/provider` already covers it, unchanged by this work), no images/alerts/refresh-nudge equivalents — the showtime-specific fields a real cinema integration will need (venue, session times) are intentionally not designed here: the pivot's optional-field, additive-only shape (point 3) lets them be added later without breaking `tmdb` or any other existing provider, once a concrete integration defines the actual need.
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply docs/specs/external-integrations.md]: Fixed in 90c6524d: every frontend branch that special-cased weather/communication external integrations (device-page, discover-page, ExternalIntegrationPage.jsx's hasDeviceScreens, install-page's redirect + install info text, and the catalog card type computation in routes/integration/index.js) now handles movies the same way. It's no longer treated as a generic device integration anywhere in the UI.

File: server/lib/external-integration/externalIntegration.validateManifest.js [hunk 23]
```diff
@@ -34,7 +34,7 @@ const {
 // These rules are the exact mirror of the canonical manifest schema owned by
 // GladysAssistant/integration-store (vendored copy in manifest.schema.json):
 // a manifest accepted by the indexer must always install here, and vice versa.
-const MANIFEST_TYPES = ['device', 'communication', 'weather'];
+const MANIFEST_TYPES = ['device', 'communication', 'weather', 'movies'];
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply server/lib/external-integration/externalIntegration.validateManifest.js:37]: Fixed in 90c6524d: added 'movies' to the enum in the vendored manifest.schema.json, with a matching description sentence for the type. Core and indexer are back in lockstep.

File: front/src/routes/integration/index.js [hunk 24]
```diff
@@ -291,7 +291,7 @@ class Integration extends Component {
         key: `external-${integration.store_slug || integration.selector}`,
         external: true,
         externalInstalled: true,
-        type: ['communication', 'weather'].includes(manifest.type) ? manifest.type : 'device',
+        type: ['communication', 'weather', 'movies'].includes(manifest.type) ? manifest.type : 'device',
```
[the remark this answers is not in the document]
Comment (vincentBesseau) [reply front/src/routes/integration/index.js:294]: Good catch, thanks for confirming it's not exploitable today. Fixed in e3875819: added 'movies' to HIDDEN_TYPES_FOR_NON_ADMIN_USERS, matching weather's rationale (shared/household provider, not a per-user account). Also updated the stale sentence in docs/specs/integration-catalog-categories.md (line 23) that still said 'not communication/weather'.

**Comment by Pierre-Gilles** [comment]
Salut @vincentBesseau,

Merci pour ce gros travail. Je tiens à le dire d'emblée : techniquement, la PR est propre. Les 35 threads de review sont résolus, la CI est verte, la couverture est à 100 %, la spec est mise à jour. Ce n'est pas la qualité qui me pose problème, c'est le périmètre.

**Le cinéma n'est pas dans la mission de Gladys.** Gladys est un assistant domotique. La météo a sa place dans le core parce qu'elle alimente les scènes, l'assistant et le quotidien de la maison. Les sorties ciné, non. Et cette PR ne se contente pas d'ajouter un widget : elle ajoute une lib core (`lib/premieres`), un service natif `tmdb`, un type de box, un trigger de scène, un job planifié, deux routes API, un nouveau type d'intégration externe avec son champ de manifest, un format pivot à garder compatible à vie et une section B.19 dans la spec. Chacun de ces éléments est une surface que je devrai maintenir seul, pour un sujet qui n'a jamais été sur ma roadmap. Je préfère être honnête plutôt que de laisser la PR s'enliser.

**Le vrai problème, c'est le précédent.** Si on accepte "un type d'intégration externe par domaine", la même PR arrivera pour les recettes, les horaires de transport, les résultats sportifs, les cours de bourse. À chaque fois une lib, un pivot, un widget, un trigger, une spec. C'est précisément le goulot d'étranglement que le système d'intégrations externes a été conçu pour supprimer.

**Ce que je propose à la place.** Tu as déjà fait la moitié du chemin en calquant le modèle météo et en préparant UGC, Pathé et CGR sur le store. Le blocage, c'est que le core n'offre aujourd'hui aucun moyen à une intégration externe d'afficher du contenu sur le dashboard sans un type dédié dans le core. C'est exactement le sujet de cette demande de fonctionnalité que j'ai ouverte le mois dernier : [Permettre aux intégrations de déclarer leurs propres widgets de dashboard (schéma JSON)](``https://community.gladysassistant.com/t/permettre-aux-integrations-de-declarer-leurs-propres-widgets-de-dashboard-schema-json/10641).`` L'idée : l'intégration décrit **quoi** afficher (valeurs, listes, cartes, jauges, boutons…) dans un schéma JSON déclaratif, comme elle le fait déjà pour son formulaire de configuration, et c'est Gladys qui décide **comment** l'afficher. Pas de HTML, pas d'iframe : le core garantit la cohérence visuelle, le dark mode et le responsive pour tous les widgets, y compris tiers.

Avec ce mécanisme, ta grille d'affiches devient un widget "liste de cartes" (image, titre, date, lien, détails) déclaré par une intégration TMDB externe, et UGC, Pathé ou CGR le réutilisent tels quels. Zéro code cinéma dans le core, et le même mécanisme sert aussi à la production solaire, au robot aspirateur ou au planning de charge de la voiture.

Je ferme donc cette PR en l'état, mais je serais très content que tu portes ce chantier générique si ça te tente : spec d'abord dans `docs/specs/external-integrations.md`, comme le veut la règle du document, puis le code. Une bonne partie de ce que tu as écrit (normalisation et bornage du payload, boucle de providers, widget, tests) se transpose presque directement. On peut en discuter sur le sujet forum avant que tu te lances, ça permettra aussi de recueillir d'autres besoins.

**Une seule condition si tu te lances :** ce chantier doit être mené avec Fable 5. C'est un changement structurel du core et du contrat des intégrations externes, et sur un sujet aussi sensible je n'accepterai aucun autre modèle. Je veux bien accompagner et relire, mais je ne reviewerai pas une spec ou un code produits autrement.

Encore merci pour l'investissement, et désolé de ne pas avoir tranché plus tôt sur le périmètre.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by Pierre-Gilles** [comment]
Le lien vers la demande de fonctionnalité a été mal formaté dans le message ci-dessus, le voici en clair :

`https://community.gladysassistant.com/t/permettre-aux-integrations-de-declarer-leurs-propres-widgets-de-dashboard-schema-json`/10641

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by vincentBesseau** [comment]
Je te laisserai donc ten occupé merci 💪
